src/main/java/com/exercise/sfe/configuration/DatasourceConfiguration.java

```diff
@@ -0,0 +1,39 @@
+package com.exercise.sfe.configuration;
+
+import javax.sql.DataSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.transaction.TransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+
+@Configuration
+@ComponentScan("com.exercise.sfe.repository")
+@EnableTransactionManagement
+public class DatasourceConfiguration {
+
+  @Bean
+  public DataSource dataSource() {
+    DriverManagerDataSource dataSource = new DriverManagerDataSource();
+    dataSource.setDriverClassName("org.postgresql.Driver");
+    dataSource.setUrl("jdbc:postgresql://localhost:5432/gift");
+    dataSource.setUsername("postgres");
+    dataSource.setPassword("zsbldqpk56");
+
+    return dataSource;
+  }
+
+  @Bean
+  public JdbcTemplate jdbcTemplate(DataSource dataSource) {
```

Comment:
Do you know all ways to create beans in Spring? Do you know the difference between public, package-private, protected and private on bean factory methods?